File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/Application.scala
package pl.edu.agh.multiscalemodelling.engine

import com.badlogic.gdx.Game
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import pl.edu.agh.multiscalemodelling.engine.logic.Board

abstract class Application extends Game {
  var spriteBatch: SpriteBatch = _
  var board: Board = _

  override def create(): Unit = spriteBatch = new SpriteBatch

  override def render(): Unit = super.render()

  override def dispose(): Unit = {
    super.dispose()
    spriteBatch.dispose()
  }

  def getSpriteBatch: SpriteBatch = spriteBatch

  def setSpriteBatch(spriteBatch: SpriteBatch): Unit = this.spriteBatch = spriteBatch
}

File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/Application.java
package pl.edu.agh.multiscalemodelling.engine;

import com.badlogic.gdx.Game;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import pl.edu.agh.multiscalemodelling.engine.logic.Board;

public abstract class Application extends Game {

    protected SpriteBatch spriteBatch;
    protected Board board;

    @Override
    public void create() {

        spriteBatch = new SpriteBatch();

    }

    @Override
    public void render() {

        super.render();

    }

    @Override
    public void dispose() {

        super.dispose();
        spriteBatch.dispose();

    }

    public SpriteBatch getSpriteBatch() {
        return spriteBatch;
    }

    public void setSpriteBatch(SpriteBatch spriteBatch) {
        this.spriteBatch = spriteBatch;
    }

}
